security/pfSense-pkg-snort/files/usr/local/www/snort/snort_rulesets.php

```diff
@@ -138,50 +135,52 @@
 	else
 		$enabled_items = $_POST['toenable'];
 
-	$a_nat[$id]['rulesets'] = $enabled_items;
+	$a_nat['rulesets'] = $enabled_items;
 
 	if ($_POST['autoflowbitrules'] == "on")
-		$a_nat[$id]['autoflowbitrules'] = 'on';
+		$a_nat['autoflowbitrules'] = 'on';
 	else {
-		$a_nat[$id]['autoflowbitrules'] = 'off';
+		$a_nat['autoflowbitrules'] = 'off';
 		if (file_exists("{$snortdir}/snort_{$snort_uuid}_{$if_real}/rules/{$flowbit_rules_file}"))
 			unlink_if_exists("{$snortdir}/snort_{$snort_uuid}_{$if_real}/rules/{$flowbit_rules_file}");
 	}
 
-	write_config("Snort pkg: save enabled rule categories for {$a_nat[$id]['interface']}.");
+	// Write updated configuration
+	config_set_path("installedpackages/snortglobal/rule/{$id}", $a_nat);
+	write_config("Snort pkg: save enabled rule categories for {$a_nat['interface']}.");
 
 	/*************************************************/
 	/* Update the snort conf file and rebuild the    */
 	/* rules for this interface.                     */
 	/*************************************************/
 	$rebuild_rules = true;
-	snort_generate_conf($a_nat[$id]);
+	snort_generate_conf($a_nat);
 	$rebuild_rules = false;
 
 	/* Soft-restart Snort to live-load new rules */
-	snort_reload_config($a_nat[$id]);
+	snort_reload_config($a_nat);
 
 	$pconfig = $_POST;
 	$enabled_rulesets_array = explode("||", $enabled_items);
-	if (snort_is_running($a_nat[$id]['uuid']))
+	if (snort_is_running($a_nat['uuid']))
 		$savemsg = gettext("Snort is 'live-reloading' the new rule set.");
 
 	// Sync to configured CARP slaves if any are enabled
 	snort_sync_on_changes();
 }
 
 if (isset($_POST['unselectall'])) {
-	$a_nat[$id]['rulesets'] = "";
+	$a_nat['rulesets'] = "";
 
 	if ($_POST['ips_policy_enable'] == "on") {
-		$a_nat[$id]['ips_policy_enable'] = 'on';
-		$a_nat[$id]['ips_policy'] = $_POST['ips_policy'];
-		$a_nat[$id]['ips_policy_mode'] = $_POST['ips_policy_mode'];
+		$a_nat['ips_policy_enable'] = 'on';
```

Comment:
I don't see any calls to set_config_path() for $a_nat nor any write_config() calls, so these changes to $a_nat here and below are not committed to the runtime config or the config.xml. Is that intentional?

Reply:
Yeah, those changes to $a_nat[] really did not have to be there. The settings are saved when the user hits the SAVE button. This same code was in one other place, and also did not need to be there either. Removed it in both places.